packages/widget/src/components/amount-selector/amount-selector.ts

```diff
@@ -68,6 +61,23 @@ export class AmountSelector extends BaseComponent {
     }
   };
 
+  _onInputAmountChangeHandler = (event: Event): void => {
+    const { value } = event.target as HTMLInputElement;
+    const amount = utils.parseUnits(
+      value,
+      this.tokenBalanceController.decimals
+    );
+    this.amount = utils.formatUnits(
+      amount,
+      this.tokenBalanceController.decimals
+    );
+
+    if (!this._validateAmount(value)) return;
```

Comment:
this has to go before parsing, not sure how it keeps ending up after which causes errors if user inputs not a number

Reply:
I double-checked that, and I think the order is right if it's a numeric value that doesn't meet the criteria f.e more than the balance we still need to set it to "this.amount" to keep the input in sync. Regarding your concern, if the user inputs some non-numeric value I wrap the code in a try-catch block if the parse fails.